File: PRM392_BrainBox_v1.0/app/src/main/java/com/example/prm392_v1/data/entity/DownloadHistory.java
package com.example.prm392_v1.data.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "download_history")
public class DownloadHistory {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public int userId;
    public int targetId;
    @NonNull
    public String targetType;
    public long downloadedAt = System.currentTimeMillis();
}

File: BrainBoxAPI/BrainBoxAPI/Models/DownloadHistory.cs
namespace BrainBoxAPI.Models
{
    public class DownloadHistory
    {
        public int UserId { get; set; }

        public int TargetId { get; set; }

        public string TargetType { get; set; } 

        public long DownloadedAt { get; set; }

        public User? User { get; set; }
    }

}
